src/hotspot/share/gc/parallel/psPromotionManager.cpp

```diff
@@ -227,7 +227,8 @@ void PSPromotionManager::restore_preserved_marks() {
 }
 
 void PSPromotionManager::drain_stacks_depth(bool totally_drain) {
-  totally_drain = totally_drain || (_target_stack_size == 0);
+  const uint threshold = totally_drain ? 0
+                                       : _target_stack_size;
```

Comment:
Feel free to ignore: I would prefer to not spread very simple `?:` statements like this over multiple lines. This is imo less readable than keeping it in a single line (and `?:` statements should always be very small/short anyway).

Reply:
+1